sklearn/utils/testing.py

```diff
@@ -588,7 +588,8 @@ def is_abstract(c):
     path = sklearn.__path__
     for importer, modname, ispkg in pkgutil.walk_packages(
             path=path, prefix='sklearn.', onerror=lambda x: None):
-        if ".tests." in modname:
+        # Prevent adding *SearchCV estimators from the old grid_search module
+        if (".tests." in modname) or ("grid_search" in modname):
```

Comment:
The `GridSearchCV` from the new module (having been derived from `BasedEstimator`) is also added to the list of all estimators along with the old `GridSearchCV` (from `grid_search` module).

This is being stored as a tuple of the form `('GridSearchCV', class_obj)` and contains 2 such entries for the `GridSearchCV`... While sorting since the name is same, the next item of tuple `class_obj` which is not a comparable is tried for comparison which had raised an error...